steward/src/cork.rs

```diff
@@ -0,0 +1,145 @@
+            Err(err) => {
+                error!("cork query client connection failed: {}", err);
+                return Err(Status::new(
+                    Code::Internal,
+                    "failed to query chain to validate cellar id",
+                ));
+            }
+        };
+
+        debug!("checking if cellar ID is approved");
+        let ids = &client
+            .query_cellar_i_ds(QueryCellarIDsRequest {})
+            .await?
+            .get_ref()
+            .cellar_ids
+            .clone();
+        if !ids.contains(&request.cellar_id) {
+            debug!(
+                "cellar ID {} not approved by governance",
+                &request.cellar_id
+            );
+            return Err(Status::new(
+                Code::PermissionDenied,
+                format!(
+                    "cellar ID {} not approved by governance",
+                    &request.cellar_id
+                ),
+            ));
+        }
+
+        // Build and send cork
+        let cork = match build_cork(request).await {
+            Ok(c) => c,
+            Err(err) => {
+                warn!("failed to build cork: {}", err);
+                return Err(Status::new(Code::InvalidArgument, err.to_string()));
+            }
+        };
+        debug!("cork: {:?}", cork);
+
+        if let Err(err) = send_cork(cork).await {
+            error!("failed to submit cork: {}", err);
+            return Err(Status::new(
+                Code::Internal,
+                "failed to send cork to sommelier",
+            ));
+        }
+        debug!("sent cork!");
+
+        Ok(Response::new(SubmitResponse {}))
+    }
+}
+
+async fn build_cork(request: &SubmitRequest) -> Result<Cork, Error> {
+    cellars::validate_cellar_id(request.cellar_id.as_str())?;
+    let address = request.cellar_id.clone();
+    let contract_call_data = match request.call_data.clone() {
+        Some(call) => call,
+        None => return Err(ErrorKind::Http.context("empty contract call data").into()),
+    };
+    let encoded_call = get_encoded_call(contract_call_data)?;
+
+    Ok(Cork {
+        encoded_contract_call: encoded_call,
+        target_contract_address: address,
+    })
+}
+
+fn get_encoded_call(data: CallData) -> Result<Vec<u8>, Error> {
+    match data {
+        AaveV2Stablecoin(call) => Ok(aave_v2_stablecoin::get_encoded_call(
+            call.function
+                .expect("call data for Aave V2 Stablecoin cellar was empty"),
+        )),
+        Uniswapv3Rebalance(params) => Ok(uniswapv3::get_encoded_call(params)),
+    }
+}
+
+async fn send_cork(cork: Cork) -> Result<TxResponse, Error> {
+    let config = APP.config();
+    debug!("establishing grpc connection");
+    let contact = Contact::new(&config.cosmos.grpc, MESSAGE_TIMEOUT, &config.cosmos.prefix)?;
+
+    debug!("loading the delegate (orchestrator) key and address from config");
+    let delegate_key = &config::DELEGATE_KEY;
+    let delegate_address = &config::DELEGATE_ADDRESS;
```

Comment:
We don't really need to assign these two to variables since we're only using them once.